rero/rero-ils
# holdings: introduce holding level

* Adds holdings level between documents and items.
* Adds automatic holding creation for new items based on item type and item location.
* Adds automatic attaching of new items to a holding record if a valid holding record exists.
* Adds automatic holding deletion once last item is deleted.
* Adds automatic re-linking if necessary to a new holding when item is updated.

Co-Authored-by: Aly Badr <aly.badr@rero.ch>

# How to Test:

* bootstrap, setup
* create an item and make sure `holding` field is added to the item record
* display the created holding record at 'localhost:5000/api/holdings/XX'
* edit the item and change the item type or the location of the item
** make sure the item is re-attached to another holding.

* delete the item and make sure the holding is also deleted by checking '/api/holdings/XX'
** (if the item is the last attached to this holding)
